src/Files.Uwp/Helpers/LayoutPreferences/LayoutPrefsDb.cs

```diff
@@ -0,0 +1,143 @@
+﻿using LiteDB;
+using System;
+using System.Linq;
+using Files.Shared.Extensions;
+
+namespace Files.Uwp.Helpers.LayoutPreferences
+{
+    public class LayoutPrefsDb : IDisposable
+    {
+        private readonly LiteDatabase db;
+
+        public LayoutPrefsDb(string connection, bool shared = false)
+        {
+            db = new LiteDatabase(new ConnectionString(connection)
+            {
+                Mode = shared ? FileMode.Shared : FileMode.Exclusive
+            }, new BsonMapper() { IncludeFields = true });
+        }
+
+        public void SetPreferences(string filePath, ulong? frn, LayoutPreferences prefs)
```

Comment:
Maybe I miss a bit of context from not often working on Files but what does `frn` stand for?

Reply:
It's NTFS "File Reference Number". It stays constant even if the filesystem path of a file or folder changes. I use it to keep the preferences intact when a folder is moved or renamed.